src/Chip/Chip.js

```diff
@@ -0,0 +1,338 @@
+    onKeyboardFocus: () => {},
+    onRequestClose: () => {},
+    onTouchEnd: () => {},
+    onTouchStart: () => {},
+    onTouchTap: () => {},
+    onMouseDown: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseUp: () => {},
+    avatar: null,
+  };
+
+  static contextTypes = {muiTheme: React.PropTypes.object.isRequired};
+
+  state = {clicked: false, focused: false};
+
+  handleBlur = (event) => {
+    this.setState({clicked: false, focused: false});
+    this.props.onBlur(event);
+  };
+
+  handleCloseIconTouchTap = (event) => {
+    //Stop the event from bubbling up to the `Chip`
+    event.stopPropagation();
+    this.props.onRequestClose(event);
+  }
+
+  handleFocus= (event) => {
+    this.setState({focused: true});
+    this.props.onFocus(event);
+  };
+
+  handleKeyboardFocus = (event, keyboardFocused) => {
+    if (keyboardFocused) {
+      this.handleFocus();
+    } else if (!this.state.hovered) {
+      this.handleBlur();
+    }
+
+    this.props.onKeyboardFocus(event, keyboardFocused);
+  };
+
+  handleKeyDown = (event) => {
+    if (keycode(event) === 'backspace') {
+      event.preventDefault();
+      this.props.onRequestClose(event);
+    } else if (keycode(event) === 'enter') {
+      event.preventDefault();
+      this.props.onTouchTap(event);
+    }
+  };
+
+  handleMouseEnter = (event) => {
+    this.handleFocus();
+    this.props.onMouseEnter(event);
+  };
+
+  handleMouseDown = (event) => {
+    //only listen to left clicks
+    if (event.button === 0) {
+      event.stopPropagation();
+      this.setState({clicked: true});
+      this.props.onMouseDown(event);
+    }
+  };
+
+  handleMouseLeave = (event) => {
+    if (!this.refs.button.isKeyboardFocused()) {
+      this.handleBlur();
+    }
+    this.props.onMouseLeave(event);
+  };
+
+  handleMouseUp = (event) => {
+    this.setState({clicked: false});
+    this.props.onMouseUp(event);
+  };
+
+  handleTouchStart = (event) => {
+    event.stopPropagation();
+    this.setState({clicked: true});
+    this.props.onTouchStart(event);
+  };
+
+  handleTouchEnd = (event) => {
+    this.setState({clicked: false});
+    this.props.onTouchEnd(event);
+  };
+
+  render() {
+    const {prepareStyles} = this.context.muiTheme;
+    const styles = getStyles(this.props, this.context, this.state);
+
+    const chipStyle = prepareStyles(Object.assign(styles.chip, this.props.chipStyle));
+    const rootStyle = Object.assign(styles.root, this.props.style);
+    const labelStyle = prepareStyles(Object.assign(styles.label, this.props.labelStyle));
+
+    const {className, deletable, ...other} = this.props;
+    let {avatar, children, label} = this.props;
+
+    let deleteIcon = deletable ?
+      <DeleteIcon
+        color={styles.closeIcon.color}
+        style={styles.closeIcon}
+        onTouchTap={this.handleCloseIconTouchTap}
+      /> :
+      null;
+
+    // If the`avatar` prop isn't set, check if the first child is an Avatar
+    if (!avatar) {
```

Comment:
I think that we should only support one of the two options. I feel like the second option is better. It's providing more possibility without much overhead.